packages/targets/dydx-stark/src/transport/send.ts

```diff
@@ -0,0 +1,96 @@
+import { Transport, TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { ResponseCache } from '@chainlink/external-adapter-framework/cache/response'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import {
+  AdapterRequest,
+  AdapterResponse,
+  makeLogger,
+} from '@chainlink/external-adapter-framework/util'
+import { BaseEndpointTypes, DyDxResponse, inputParameters } from '../endpoint/send'
+import { getPricePayload, PriceDataPoint, PriceStarkPayload, requireNormalizedPrice } from './utils'
+import { calculateHttpRequestKey } from '@chainlink/external-adapter-framework/cache'
+
+const logger = makeLogger('DyDxTransport')
+
+export type DyDxTransportTypes = BaseEndpointTypes & {
+  Provider: {
+    RequestBody: PriceStarkPayload
+    ResponseBody: DyDxResponse
+  }
+}
+
+export class DyDxTransport implements Transport<DyDxTransportTypes> {
+  name!: string
+  responseCache!: ResponseCache<DyDxTransportTypes>
+  requester!: Requester
+
+  async initialize(
+    dependencies: TransportDependencies<DyDxTransportTypes>,
+    _adapterSettings: DyDxTransportTypes['Settings'],
+    _endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    this.responseCache = dependencies.responseCache
+    this.requester = dependencies.requester
+    this.name = transportName
+  }
```

Comment:
I think `name` and `responseCache` should already be accessible. We should also initialize here.
```suggestion
  requester!: Requester

  async initialize(
    dependencies: TransportDependencies<DyDxTransportTypes>,
    _adapterSettings: DyDxTransportTypes['Settings'],
    _endpointName: string,
    transportName: string,
  ): Promise<void> {
    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
    this.requester = dependencies.requester
  }
```

Reply:
The class is not inheriting from other class. It just implements the `Transport` interface. The main reason for this was to use `foregroundExecute` since this EA posts data to DP. Not sure if it makes sense to convert it to backgroundExecute, subscription based transport.

Reply:
Ah sorry, I definitely skimmed over that part. This is good then